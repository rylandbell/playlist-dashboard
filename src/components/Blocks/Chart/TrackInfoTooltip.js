import React, { Component } from 'react';

class TrackInfoTooltip extends Component {
  render() {
    const shouldRenderTooltip =
      this.props.payload && this.props.payload.length > 0;

    return (
      shouldRenderTooltip &&
      <div className="chart__track-info-tooltip">
        <p className="desc">{this.props.payload[0].payload.name}</p>
      </div>
    );
  }
}

export default TrackInfoTooltip;
