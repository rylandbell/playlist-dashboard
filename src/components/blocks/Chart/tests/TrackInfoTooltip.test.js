import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import TrackInfoTooltip from "../TrackInfoTooltip";

describe("TrackInfoTooltip component", () => {
  const props = {
    payload: [
      {
        payload: { name: "testName" }
      }
    ]
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TrackInfoTooltip {...props} />, div);
  });

  it("shallow render matches snapshot", () => {
    const tree = toJson(shallow(<TrackInfoTooltip {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
