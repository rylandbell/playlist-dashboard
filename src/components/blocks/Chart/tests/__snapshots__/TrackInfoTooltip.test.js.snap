// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackInfoTooltip component shallow render matches snapshot 1`] = `
<div
  className="chart__track-info-tooltip"
>
  <p
    className="desc"
  >
    testName
  </p>
</div>
`;
